typing/typecore.ml

```diff
@@ -778,21 +784,51 @@ let solve_constructor_annotation
         | _ -> assert false
   in
   if ids <> [] then ignore begin
+    let get_decl_manifest id env =
+      let decl =
+        try Env.find_type (Path.Pident id) env with Not_found -> assert false in
```

Comment:
Is there a reason that the code is re-querying the declaration and manifest of the `ids` types from the environment? Would it not be more straightforward to store those in the `ids`  list?

Reply:
This avoids thinking hard about the code, but indeed it is ok to keep the values along, so I changed the code.